src/editor/world_editor.cpp

```diff
@@ -289,9 +448,10 @@ class RemoveArrayPropertyItemCommand : public IEditorCommand
 		}
 
 	private:
+		WorldEditor& m_editor;
 		Component m_component;
 		int m_index;
-		IArrayDescriptor& m_descriptor;
+		const IArrayDescriptor* m_descriptor;
```

Comment:
since m_descriptor is pointer now, you should assert not null on the every place you use it.